diff --git a/opengl/DearImGui/imgui_impl_opengl2.cpp b/opengl/DearImGui/imgui_impl_opengl2.cpp
index 2a255be..3d30626 100644
--- a/opengl/DearImGui/imgui_impl_opengl2.cpp
+++ b/opengl/DearImGui/imgui_impl_opengl2.cpp
@@ -152,6 +152,7 @@ static void ImGui_ImplOpenGL2_SetupRenderState(ImDrawData* draw_data, int fb_wid
     glShadeModel(GL_SMOOTH);
     glTexEnvi(GL_TEXTURE_ENV, GL_TEXTURE_ENV_MODE, GL_MODULATE);
 
+#ifndef IMGUI_DPF_BACKEND
     // If you are using this code with non-legacy OpenGL header/contexts (which you should not, prefer using imgui_impl_opengl3.cpp!!),
     // you may need to backup/reset/restore other state, e.g. for current shader using the commented lines below.
     // (DO NOT MODIFY THIS FILE! Add the code in your calling function)
@@ -173,6 +174,11 @@ static void ImGui_ImplOpenGL2_SetupRenderState(ImDrawData* draw_data, int fb_wid
     glMatrixMode(GL_MODELVIEW);
     glPushMatrix();
     glLoadIdentity();
+#else
+    IM_UNUSED(draw_data);
+    IM_UNUSED(fb_width);
+    IM_UNUSED(fb_height);
+#endif
 }
 
 // OpenGL2 Render function.
